### DISCUSSION TITLE
How to change deckgl prop value(extruded) on button click?

### DISCUSSION THREAD

**Discussion by vivek091195** [visgl/deck.gl#7547] [Q&A]
I am plotting a hex layer in my app, and I want to modify extruded prop to true on click of a button. Is there a way to do it without plotting the layer again? I have tried to replicate the required scenario in this [fiddle](https://jsfiddle.net/vivekkumar091195/y6z9faw3/1/)

Below is the layer invocation code,
```
new H3HexagonLayer({
        id: 'h3-hexagon-layer',
        data: 'https://raw.githubusercontent.com/visgl/deck.gl-data/master/website/sf.h3cells.json',
        pickable: true,
        filled: true,
        extruded: false,
        elevationScale: 20,
        getLineColor: [0, 0, 0],
        getLineWidth: 1,
        getHexagon: d => d.hex,
        getFillColor: d => [255, (1 - d.count / 500) * 255, 0],
        getElevation: d => d.count
      }),
```

**Comment by Pessimistress** [comment]
> Is there a way to do it without plotting the layer again

No, layer props are immutable. See https://deck.gl/docs/faq#should-i-really-regenerate-all-layers-every-time-something-changes

Example: https://codepen.io/vis-gl/pen/NYYeNj
